pandas/core/indexing.py

```diff
@@ -2098,8 +2099,15 @@ def _setitem_with_indexer_missing(self, indexer, value):
                 # We should not cast, if we have object dtype because we can
                 # set timedeltas into object series
                 curr_dtype = self.obj.dtype
-                curr_dtype = getattr(curr_dtype, "numpy_dtype", curr_dtype)
```

Comment:
this is already pretty kludgy...

Reply:
Yeah, I certainly don't disagree .. But this not really new to this PR, happy to hear suggestions. 

This part of the code was added in https://github.com/pandas-dev/pandas/pull/47342/ to support retaining the dtype in setitem with expansion. 
I am wondering though if we could simplify this by "just" trying to create an array from the scalar first with the dtype of the series, and then if that fails fall back on creating an array without specifying a dtype, and then letting inference and concat/common_dtype do its thing (so without trying to determine the `new_dtype` up front)

Reply:
Something like 

```
try:
    new_values = Series([value], dtype=self.obj.dtype)._values
except ...:
    new_values = Series([value])._values
```

(although I assume there is a reason for all the checks .. Can see what test would fail with the above)

Reply:
I expect the try/except approach here would run into the same set of problems that _from_scalars is intended to address.  I'm hoping you and i can collaborate on implementing that soonish.

Maybe shorter-term something like:

```
dummy = ser.iloc[:1].copy()  #  assuming self.obj is a Series, needs an additional getitem otherwise
dummy.iloc[0] = value
```

Then see if dummy raises/casts to determine if we can keep the dtype?  This would be a little bit like #36226 but without a new method.